src/OrleansRuntime/Scheduler/WorkerPoolThread.cs

```diff
@@ -69,26 +71,25 @@ internal Task CurrentTask
             set
             {
                 currentTask = value;
-                currentWorkItem = null;
-                CurrentStateStarted = DateTime.UtcNow;
+                currentTaskStarted = DateTime.UtcNow;
```

Comment:
the main change in this PR is this line: do not reset `currentWorkItem` to null when setting `CurrentTask`. That will preserve the current work item for the case of work item group (activation data) when setting the individual `Task` for that work item group. This should allow us to print currentWorkItem  when finding a frozen individual `Task`.